standardization/src/main/scala/za/co/absa/enceladus/standardization/interpreter/stages/TypeParser.scala

```diff
@@ -48,7 +54,7 @@ import scala.util.{Random, Try}
   *         TimestampParser !
   *         DateParser !
   */
-sealed trait TypeParser {
+sealed trait TypeParser[T] {
```

Comment:
I wondered, wouldn't be better to rename this class to `TypeCaster`?
Closer to it's actual function, and would avoid class name clashes for `IntegralParser` etc. (not a real issues, just little confusing). WDYT?

Reply:
Sounds reasonable, but I would do it in a separate PR, this is already too big.

Reply:
That makes sense 👍